modules/sor/sorV2/sorV2.service.ts

```diff
@@ -0,0 +1,243 @@
+import { 
+    BasePool, 
+    ChainId,  
+    sorGetSwapsWithPools, 
+    Token, 
+    Address, 
+    SwapKind, 
+    sorParseRawPools, 
+    RawStablePool, 
+    RawLinearPool, 
+    RawWeightedPool, 
+    RawComposableStablePool, 
+    RawMetaStablePool,
+    Swap
+} from '@balancer/sdk';
+import { GqlSorGetSwapsResponseNew, GqlSorSwapType } from '../../../schema';
+import { PrismaPoolType, PrismaToken } from '@prisma/client';
+import { GetSwapsInput } from '../sor.service';
+import { tokenService } from '../../token/token.service';
+import { networkContext } from '../../network/network-context.service';
+import { prisma } from '../../../prisma/prisma-client';
+import { PrismaPoolWithDynamic, prismaPoolWithDynamic } from '../../../prisma/prisma-types';
+import { RawPool } from '@balancer/sdk';
+import { HumanAmount, SupportedRawPoolTypes } from '@balancer/sdk';
+
+import { getSwapCompare } from './temp';
+
+// TODO - Check if this has been deployed to same address across networks?
+export const SOR_QUERIES = '0x1814a3b3e4362caf4eb54cd85b82d39bd7b34e41';
+
+export class SorV2Service {
+    public async getSwaps({
+        tokenIn,
+        tokenOut,
+        swapType,
+        swapAmount,
+    }: GetSwapsInput): Promise<GqlSorGetSwapsResponseNew> {
+        // TODO - This takes ~1.262s on my local machine. Is this fast enough? Any obvious ways to improve?
+        console.time('poolsFromDb');
+        const poolsFromDb = await this.getBasePoolsFromDb();
```

Comment:
Not sure how representative the timing from my local machine is compared to production?
Maybe I'm not approaching this correctly. Is there any obvious ways this can be improved?